internal/services/computefleet/compute_fleet_resource.go

```diff
@@ -890,29 +876,31 @@ func (r ComputeFleetResource) Update() sdk.ResourceFunc {
 
 			// API requires `osProfile.adminPassword` when updating resource but the GET API does not return the sensitive data `osProfile.adminPassword`
 			if props := properties.Properties; props != nil {
-				if len(model.VirtualMachineProfile[0].OsProfile[0].LinuxConfiguration) > 0 {
-					if v := props.ComputeProfile.BaseVirtualMachineProfile.OsProfile; v != nil {
+				if v := props.ComputeProfile.BaseVirtualMachineProfile.OsProfile; v != nil {
```

Comment:
what if `ComputeProfile == nil` or `BaseVirtualMachineProfile == nil` ?

Reply:
`ComputeProfile` and `BaseVirtualMachineProfile` are not pointer types, so I assume they will not be equal to nil.